### PR TITLE
WIP: chat nav2 headers

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] plumbing
- [x] search
- [ ] pending

### CODE REVIEW

File: shared/chat/conversation/header-area/container.js [hunk 1]
```diff
@@ -24,6 +24,7 @@ type Props = {|
 
 class HeaderArea extends React.PureComponent<Props> {
   render() {
+    // TODO handle this in new router
```
Comment (chrisnojima) [inline shared/chat/conversation/header-area/container.js]: this isn't handled yet

**Comment by cjb** [comment]
@keybase/react-hackers I think this is ready now!  Added Search and Mute to the desktop header, and retained the nav1 header on mobile for now.

File: shared/chat/conversation-filter-input.js [hunk 2]
```diff
@@ -212,12 +212,18 @@ const styles = Styles.styleSheetCreate({
...
   containerNotFiltering: Styles.platformStyles({
     common: {
-      height: 48,
+      height: flags.useNewRouter ? undefined : 48,
       position: 'relative',
     },
     isElectron: {
       ...Styles.padding(0, Styles.globalMargins.xtiny),
       backgroundColor: Styles.globalColors.blueGrey,
+      ...(flags.useNewRouter
```
Comment (buoyad) [inline shared/chat/conversation-filter-input.js]: I think cleaner to wrap the whole object in this check

File: shared/chat/header.desktop.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+              <Kb.Icon
+                type="iconfont-shh"
+                style={styles.shhIconStyle}
+                color={Styles.globalColors.black_20}
+                fontSize={20}
+                onClick={p.unMuteConversation}
+              />
+            )}
+          </Kb.Box2>
+          {!!p.desc && <Kb.Text type="BodyTiny">{p.desc}</Kb.Text>}
```
Comment (buoyad) [inline shared/chat/header.desktop.js]: Long descriptions push the icons off the right of the window

File: shared/router-v2/header/index.desktop.js [hunk 4]
```diff
@@ -103,6 +103,7 @@ const styles = Styles.styleSheetCreate({
     isElectron: {
       ...Styles.desktopStyles.windowDragging,
       alignItems: 'center',
+      maxHeight: 80,
```
Comment (buoyad) [inline shared/router-v2/header/index.desktop.js]: :fire: busts team headers

File: shared/chat/header.desktop.js [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+  const meta = stateProps._meta
+  return {
+    channel:
+      meta.teamType === 'big'
+        ? `${meta.teamname}#${meta.channelname}`
+        : meta.teamType === 'small'
+        ? meta.teamname
+        : null,
+    desc: meta.description,
+    infoPanelOpen: false, // not really needed stateProps.infoPanelOpen,
```
Comment (buoyad) [inline shared/chat/header.desktop.js]: Remove this then? :sheep: personally I think it becoming an (x) is nice

**Comment by cjb** [comment]
@ayoubd Ready for another look! Put the infoPanelOpen logic into Redux since I wasn't getting a callback when the popup closed. Couldn't repro descriptions doing the wrong thing when too long with the latest commit.

File: shared/constants/chat2/index.js [hunk 6]
```diff
@@ -171,8 +172,12 @@ export const isTeamConversationSelected = (state: TypedState, teamname: string)
   return meta.teamname === teamname
 }
 export const isInfoPanelOpen = (state: TypedState) => {
-  const routePath = getPath(state.routeTree.routeState, [chatTab])
-  return routePath.size === 3 && routePath.get(2) === 'chatInfoPanel'
+  if (flags.useNewRouter) {
+    return state.chat2.infoPanelOpen
```
Comment (buoyad) [inline shared/constants/chat2/index.js]: you can derive this from the stack with new router too, teams does [something similar](https://github.com/keybase/client/blob/master/shared/constants/teams.js#L297)

Comment (adamjspooner) [reply shared/constants/chat2/index.js]: I haven't checked out this branch, but is it known that the “Jump to...” and new conversation button are on a non-white background?

Comment (buoyad) [reply shared/constants/chat2/index.js]: I didn't notice that @adamjspooner good eye!

**Comment by cjb** [comment]
@buoyad @adamjspooner Fixed the unnecessary Reduxing of infoPanelOpen and the grey background!

I haven't fixed the right-side icons going off-screen -- you're right, it only does that when there are no spaces in the description.  I spent some time trying to move to ellipsis/overflow to solve it, but there seem to be gotchas with the flexboxing that I haven't been able to work out so far.  Since having a description with no spaces is an edge case, perhaps best to come back to this during the design QA on these?

**Pull request merged**